openedx/core/djangoapps/course_apps/tasks.py

```diff
@@ -0,0 +1,57 @@
+"""
+This file contains celery tasks for course apps.
+"""
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+from edx_django_utils.monitoring import set_code_owner_attribute
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locator import LibraryLocator
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.course_apps.models import CourseAppStatus
+from openedx.core.djangoapps.course_apps.plugins import CourseAppsPluginManager
+
+log = get_task_logger(__name__)
+
+
+@shared_task(name='openedx.core.djangoapps.course_apps.tasks.cache_all_course_apps_status')
+@set_code_owner_attribute
+def cache_all_course_apps_status():
+    """
+    Create CourseAppStatus entries for all course apps, across all courses to speed up queries.
+    """
+    course_app_plugins = CourseAppsPluginManager.get_available_plugins()
+    for idx, course_key in enumerate(CourseOverview.objects.values_list('id', flat=True)):
+        if (idx + 1) % 100 == 0:
+            log.info("Cached %s records successfully", idx + 1)
```

Comment:
Optional: You might want to consider logging the CourseKey at the start, to make it easier to debug if you encounter a weird edge case.

Reply:
Sure, I was worried that would make the output too verbose, but it is just supposed to run once.